src/wipplpy/modules/connection.py

```diff
@@ -0,0 +1,43 @@
+"""
+Define the generic superclass through which to make MDSplus connections.
+
+TODO: Add a MySQL connection and maybe other connection types as well?
+"""
+
+# Standard
+import socket
+
+
+class MDSPlusConnection:
+
+    mdsplus_server = None
+    tree_name = None
+
+    def __init__(self, ...):
+        self.is_remote_connection = None
+
+    # Abstract method
```

Comment:
This should use the `@abstractmethod` method from the `abc` library. See https://stackoverflow.com/questions/13646245/is-it-possible-to-make-abstract-classes-in-python.